app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -363,7 +369,8 @@ def after_create_response
     def after_update_response
       if concern_has_file_sets?
         return redirect_to hyrax.confirm_access_permission_path(curation_concern) if permissions_changed?
-        return redirect_to main_app.confirm_hyrax_permission_path(curation_concern) if curation_concern.visibility_changed?
+        # is this codepath unreachable? how can visibility have changed if permissions did not?!
```

Comment:
looking at this a bit closer, i'm pretty sure this is only reachable in two cases:

  - when the test suite has stubbed `visibility_changed?` (which seems to be the case whenever it's actually hit during testing, i'm running the full suite to double check this); and
  - when the `visibility_will_change!` flag has been set, but the permissions (and visibility has not actually changed)
    - e.g. if i set `visibility = 'public'; visibility = 'restricted'` on a work that started as "restricted", i'll have set the flag, but `premissions_changed?` will be false. i don't think we want to be redirecting in this case, and i strongly suspect this code, and the whole controller action/view that follow from it have been unreachable since the above line was added in 2017.

Reply:
a fix for this smaller issue is now in at #4650

Reply:
I would imagine so; My recollection is of a one time quasi-separation of visibility and permission, but really they are conflated so this likely was added. I think your assessment is spot on and accurate.